core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
> I think I am implicitly reasoning that the file creation, for example, must happen-before the assignment within one thread (this is not a question of the Java memory model and visibility).

For this one, because I have not seen any docs about it before, I'm not sure it's really guaranteed.